drivers/stepper/adi_tmc/bus/adi_tmc_uart.c

```diff
@@ -0,0 +1,163 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025 Dipak Shetty
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/sys/util.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/byteorder.h>
+#include <adi_tmc_uart.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(tmc_uart, CONFIG_STEPPER_LOG_LEVEL);
+
+/* TMC UART standard datagram size */
+#define ADI_TMC_UART_DATAGRAM_SIZE          8
+/* TMC UART read request datagram size */
+#define ADI_TMC_UART_READ_REQ_DATAGRAM_SIZE 4
+/* TMC UART protocol constants */
+#define ADI_TMC_UART_SYNC_BYTE              0x05
+#define ADI_TMC_UART_WRITE_BIT              0x80
+
+uint8_t tmc_uart_calc_crc(const uint8_t *datagram, uint8_t len)
```

Comment:
Can the common crc routine in `lib/crc` be used?

Reply:
Tested it with the library crc functions, and they do not seem to match. This seems to be a custom CRC calculation.